### PR TITLE
[IN PROGRESS] Update to pivot functions

### PR DESCRIPTION
**Pull request by mine-cetinkaya-rundel**
What I've already done:

- `spread`/`gather` -> `pivot_*`
- Fix a few typos
- Add missing links to other primers
- Update one slightly confusing MC question

What we need to make a decision on before PR can be merged:

- There was one note in the existing primer "[To be replaced with a video]". I haven't taken that out. We can make the video or take out that note.
- There are two existing videos (search for "vimeo" in the document) that will need to be updated. I imagine @garrettgman has the keynote slides that go with them. I'd be happy to update if you can share the Keynote slides, or we can hold off until you can make the videos and merge the PR then.

### CODE REVIEW

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 1]
```diff
@@ -269,46 +269,50 @@ question("What are the variables in cases?",
...
+Let's take a closer look at the `pivot_longer()` syntax.
 
-### gather() syntax
+### pivot_longer() syntax
 
 Here's the same call written without the pipe operator, which makes the syntax easier to see.
 
 ```{r echo = TRUE, eval = FALSE}
-gather(cases, key = "year", value = "n", 2, 3, 4)
+pivot_longer(cases, cols = c(`2011`, `2012`, `2013`), names_to = "year", values_to = "n")
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: The `cols` argument here lists the actual years, whereas the previous chunk uses `-Country` to select columns. Should they be consistent? The text reads that it is the same call written without the pipe operator, so it might be simpler to only have the inclusion of `cases %>%` be the thing that is different here.

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 2]
```diff
@@ -269,46 +269,50 @@ question("What are the variables in cases?",
...
 ```
 
-To use `gather()`, pass it the name of a data set to reshape followed by two new column names to use. Each name should be a character string surrounded by quotes: 
+To use `pivot_longer()`, pass it the name of a data set to reshape followed by which columns to pivot longer, the name of a new variable that will contain on the names of these columns as values, and the name of another new variables that will contain the values from these columns:
 
-* the key string will become the name of a new column that contains former column names. 
-* the value string will become the name of a new column that contains former cell values.
+* the `cols` argument contains the name of the columns to pivot into longer format.
+* the `names_to` argument is a string specifying the name of the column to create from the data stored in the column names of the dataset to be reshaped.
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: May be redundant, but I like "string specifying the name of the *new* column to create..."

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 3]
```diff
@@ -269,46 +269,50 @@ question("What are the variables in cases?",
...
+* the `values_to` argument is a string specifying the name of the column to create from the data stored in cell values.
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: same, "new column to create"

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd (hunk 1 above)
Comment (apreshill) [reply tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: Just got down the "specifying columns" section- perhaps suggests we should keep this example as is with the years named....

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 4]
```diff
@@ -269,46 +269,50 @@ question("What are the variables in cases?",
...
 
-### gather()
+### pivot_longer()
 
-You can use the `gather()` function in the **tidyr** package to convert wide data to long data. Notice that `gather()` returns a tidy copy of the dataset, but does not alter the original dataset. If you wish to use this copy later, you'll need to save it somewhere.
+You can use the `pivot_longer()` function in the **tidyr** package to convert wide data to long data. Notice that `pivot_longer()` returns a tidy copy of the dataset, but does not alter the original dataset. If you wish to use this copy later, you'll need to save it somewhere.
 
 ```{r echo = TRUE}
-cases %>% gather(key = "year", value = "n", 2, 3, 4)
+cases %>%
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: Change this code to match below:
```
cases %>% pivot_longer(cols = c(`2011`, `2012`, `2013`), names_to = "year", values_to = "n")
```

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 5]
```diff
@@ -330,20 +334,19 @@ table4a %>% gather(key = "year", value = "cases", 2, 3)
 
 ### Specifying columns
 
-So far we've used numbers to describe which columns to reshape with `gather()`, but this isn't necessary. `gather()` also recognizes column names as well as all of the `select()` helpers that you learned about in [Isolating Data with dplyr](). So for example, these expressions would all do the same thing:
+So far we have specified explicitly which columns to pivot, but this isn't necessary. `pivot_longer()` also recognizes column names as well as all of the `select()` helpers that you learned about in [Isolating Data with dplyr](https://rstudio.cloud/learn/primers/2.2). So, for example, these expressions would all do the same thing:
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: Suggest simpler words, maybe like: "So far, we have listed which columns to pivot by naming them one at a time and combining them using the `c()` function, but this isn't necessary."

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 6]
```diff
@@ -363,23 +371,34 @@ table4b %>% gather(key = "year", value = "population", -country)
 
 ### Converting output
 
-If you looked closely at your results in the previous exercises, you may have noticed something odd: the new year column contains character vectors. You can tell because R displays `<chr>` beneath the column name.
+If you looked closely at your results in the previous exercises, you may have noticed something odd: the new year column is a character vector. You can tell because R displays `<chr>` beneath the column name. `names_ptypes` and `values_ptypes` arguments take a list of column name-prototype (ptype) pairs defining the desired type of each newly created column.
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: This syntax seems a bit hard compared to the previous way :sob:. I feel like we need to include that each item within the `list()` consists of the bare variable name from the *resulting* dataset on the left (so, we created `year` above in quotes, now it is not in quotes) and on the right the variable type (how do we know `integer()` is an option? I know they are in base, but not totally intuitive. And can we steer them to a help page with the other options?)

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 7]
```diff
@@ -392,53 +411,59 @@ Rectangular tables are a simple form of wide data. But you will also encounter m
 cases2
 ```
 
-To tidy this data, you would want to keep the first three columns as they are. Can you tidy this data with `gather()`? Yes, an you already know how. Think about the problem and then tidy `cases2` into a data set with five columns: `city`, `country`, `continent`, `year`, and `cases`.
+To tidy this data, you would want to keep the first three columns as they are. Can you tidy this data with `pivot_longer()`? Yes, and you already know how. Think about the problem and then tidy `cases2` into a data set with five columns: `city`, `country`, `continent`, `year`, and `cases`.
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: Edit: `year` (as an integer)

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 8]
```diff
@@ -450,19 +475,23 @@ question("Which column in pollution contains the values associated with the key
...
-pollution %>% spread(key = size, value = amount)
+pollution %>%
+  pivot_wider(
+    names_from = size,
+    values_from = amount
+  )
 ```
 
-`spread()` will give each unique value in the _key_ column its own column. The name of the value will become the column name. `spread()` will then redistribute the values in the _value_ column across the new columns in a way that preserves every relationship in the original dataset. 
+`pivot_wider()` will give each unique value in the `names_from` column its own column. The values from this column will become column names. `pivot_wider() will then redistribute the values in the `values_from` column across the new columns in a way that preserves every relationship in the original dataset.
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: "The *distinct* (or unique?) values from this column will become the *new* column names"?

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 9]
```diff
@@ -450,19 +475,23 @@ question("Which column in pollution contains the values associated with the key
...
+  )
 ```
 
-`spread()` will give each unique value in the _key_ column its own column. The name of the value will become the column name. `spread()` will then redistribute the values in the _value_ column across the new columns in a way that preserves every relationship in the original dataset. 
+`pivot_wider()` will give each unique value in the `names_from` column its own column. The values from this column will become column names. `pivot_wider() will then redistribute the values in the `values_from` column across the new columns in a way that preserves every relationship in the original dataset. 
 
 ### Exercise 3 - Tidy table2
 
-Use `spread()` to tidy `table2` into a dataset with four columns: `country`, `year`, `cases`, and `population`. In short, convert `table2` to look like `table1`.
+Use `pivot_wider()` to tidy `table2` into a dataset with four columns: `country`, `year`, `cases`, and `population`. In short, convert `table2` to look like `table1`.
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: More of a style note as I am not looking at the rendered exercises, would it be helpful to list anything with 2 or more elements (like the column names) in a md bullet or numbered list, instead of as text within backticks? I find it a bit easier to read, but I may be odd. So:

Use `pivot_wider()` to tidy `table2` into a dataset with four columns: 

+ `country`, 
+ `year`, 
+ `cases`, and 
+ `population`. 

In short, convert `table2` to look like `table1`.

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 10]
```diff
@@ -474,21 +503,25 @@ table2
...
+table2 %>% 
+  pivot_wider(
+    names_from = type, 
+    values_from = count
+    )
 ```
 
 ```{r ex11-check}
-"Good job! You now posses two complementary tools for reshaping the layout of data. By iterating between gather() and spread() you can rearrange the values of any data set into many different configurations."
+"Good job! You now posses two complementary tools for reshaping the layout of data. By iterating between pivot_longer() and pivot_wider() you can rearrange the values of any data set into many different configurations."
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: Typo: posses -> possess

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd [hunk 11]
```diff
@@ -621,10 +657,13 @@ babynames %>%
 babynames %>%
   group_by(year, sex) %>% 
   summarise(total = sum(n)) %>%
```
Comment (apreshill) [inline tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: I'm guessing previous primers covered this and not `count` 😇

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd (hunk 1 above)
Comment (mine-cetinkaya-rundel) [reply tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: Agreed + fixed

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd (hunk 2 above)
Comment (mine-cetinkaya-rundel) [reply tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: 👍

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd (hunk 3 above)
Comment (mine-cetinkaya-rundel) [reply tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: 👍

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd (hunk 5 above)
Comment (mine-cetinkaya-rundel) [reply tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: 👍

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd (hunk 7 above)
Comment (mine-cetinkaya-rundel) [reply tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: 👍

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd (hunk 9 above)
Comment (mine-cetinkaya-rundel) [reply tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: I think with the styling in the rendered document it looks fine and more concise.

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd (hunk 10 above)
Comment (mine-cetinkaya-rundel) [reply tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: 👍

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd (hunk 11 above)
Comment (mine-cetinkaya-rundel) [reply tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: I'm going to leave this one as is to not be changing multiple aspects at this time. Happy to do a review of count/summarise of all primers if we decide that's important.

File: tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd (hunk 6 above)
Comment (mine-cetinkaya-rundel) [reply tidy-data/01-Reshape-Data/01-Reshape-Data.Rmd]: I added a bit more detail..
